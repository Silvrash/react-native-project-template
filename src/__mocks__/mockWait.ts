export async function mockWait(timeout: number) {
	return new Promise(resolve => {
		setTimeout(() => {
			resolve(true);
		}, timeout);
	});
}
