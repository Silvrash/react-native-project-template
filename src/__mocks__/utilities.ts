import { useNavigation } from '@react-navigation/native';

export function setupNavigationMock() {
	const navigate = jest.fn();
	const goBack = jest.fn();
	const dispatch = jest.fn();
	const removeListener = jest.fn();
	const addListener = jest.fn();

	(useNavigation as jest.Mock).mockReturnValue({
		navigate,
		goBack,
		dispatch,
		addListener,
		removeListener,
	});
	addListener.mockImplementation((_key: string, fn: (e: any) => void) => {
		fn({ preventDefault: () => null });
	});
}
