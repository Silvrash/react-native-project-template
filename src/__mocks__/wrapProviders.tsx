import { render, RenderOptions } from '@testing-library/react-native';
import { PropsWithChildren } from 'react';
import NavigationWrapper from 'src/wrapper.navigation';
import ThemeWrapper from '../wrapper.theme';

const AllTheProviders = ({ children }: PropsWithChildren) => {
	return (
		<ThemeWrapper>
			<NavigationWrapper>{children}</NavigationWrapper>
		</ThemeWrapper>
	);
};

export function renderWithProviders<T>(ui: React.ReactElement<T>, options?: RenderOptions) {
	return render(ui, {
		wrapper: ({ children }) => <AllTheProviders>{children}</AllTheProviders>,
		...options,
	});
}
