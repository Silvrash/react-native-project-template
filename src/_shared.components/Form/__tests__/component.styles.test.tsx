import { renderWithProviders } from 'src/__mocks__/wrapProviders';
import { ErrorText, HelperText, Label } from '../component.styles';

describe('Form components', () => {
	it('should render helper text', () => {
		const { toJSON } = renderWithProviders(<HelperText />);
		expect(toJSON()).toMatchSnapshot();
	});

	it('should render helper text', () => {
		const { toJSON } = renderWithProviders(<ErrorText />);
		expect(toJSON()).toMatchSnapshot();
	});

	it('should render helper text', () => {
		const { toJSON } = renderWithProviders(<Label />);
		expect(toJSON()).toMatchSnapshot();
	});
});
