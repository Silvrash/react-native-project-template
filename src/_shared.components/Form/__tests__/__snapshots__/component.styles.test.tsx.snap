// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form components should render helper text 1`] = `
<Text
  dataSet={{}}
  style={
    {
      "backgroundColor": undefined,
      "color": "#4C4D6E",
      "fontFamily": "DMSans-Regular",
      "fontSize": 11,
      "fontStyle": undefined,
      "fontWeight": undefined,
      "letterSpacing": 0,
      "lineHeight": 15,
      "textDecorationLine": undefined,
    }
  }
/>
`;

exports[`Form components should render helper text 2`] = `
<Text
  dataSet={{}}
  style={
    {
      "backgroundColor": undefined,
      "color": "#f87171",
      "fontFamily": "DMSans-Medium",
      "fontSize": 12,
      "fontStyle": undefined,
      "fontWeight": undefined,
      "letterSpacing": 0,
      "lineHeight": 18,
      "textDecorationLine": undefined,
    }
  }
/>
`;

exports[`Form components should render helper text 3`] = `
<Text
  dataSet={{}}
  style={
    {
      "backgroundColor": undefined,
      "color": "#4C4D6E",
      "fontFamily": "DMSans-Bold",
      "fontSize": 14,
      "fontStyle": undefined,
      "fontWeight": undefined,
      "letterSpacing": 0,
      "lineHeight": 21,
      "marginTop": 8,
      "textDecorationLine": undefined,
    }
  }
/>
`;
